src/internal/operations/manifests.go

```diff
@@ -43,24 +43,38 @@ func produceManifestsAndMetadata(
 	ctx context.Context,
 	mr manifestRestorer,
 	gb getBackuper,
-	reasons []kopia.Reason,
+	reasons, fallbackReasons []kopia.Reason,
 	tenantID string,
 	getMetadata bool,
 	errs *fault.Bus,
 ) ([]*kopia.ManifestEntry, []data.RestoreCollection, bool, error) {
 	var (
+		tags          = map[string]string{kopia.TagBackupCategory: ""}
 		metadataFiles = graph.AllMetadataFileNames()
 		collections   []data.RestoreCollection
+		fbms          []*kopia.ManifestEntry
 	)
 
-	ms, err := mr.FetchPrevSnapshotManifests(
-		ctx,
-		reasons,
-		map[string]string{kopia.TagBackupCategory: ""})
+	ms, err := mr.FetchPrevSnapshotManifests(ctx, reasons, tags)
 	if err != nil {
-		return nil, nil, false, err
+		return nil, nil, false, clues.Wrap(err, "looking up prior snapshots")
+	}
+
+	if len(fallbackReasons) > 0 {
```

Comment:
do we need a len check? Ideally the code to find previous manifests should return nothing if no reasons are given

Reply:
guess not!